File: unload-manager/src/main/java/com/starship/unloadmanager/exception/ResourceException.java
package com.starship.unloadmanager.exception;

import org.springframework.http.HttpStatus;

public class ResourceException extends RuntimeException {
    private final HttpStatus httpStatus;

    public ResourceException(HttpStatus httpStatus) {
        this.httpStatus = httpStatus;
    }

    public ResourceException(HttpStatus httpStatus, String message) {
        super(message);
        this.httpStatus = httpStatus;
    }

    public HttpStatus getHttpStatus() {
        return httpStatus;
    }
}

File: unload-manager/src/main/kotlin/com/starship/unloadmanager/exception/ResourceException.kt
package com.starship.unloadmanager.exception

import org.springframework.http.HttpStatus

open class ResourceException : RuntimeException {
    val httpStatus: HttpStatus

    constructor(httpStatus: HttpStatus) {
        this.httpStatus = httpStatus
    }

    constructor(httpStatus: HttpStatus, message: String?) : super(message) {
        this.httpStatus = httpStatus
    }
}
